node/node_test.go

```diff
@@ -468,6 +468,62 @@ func TestNodeNewNodeCustomReactors(t *testing.T) {
 	assert.Contains(t, channels, cr.Channels[0].ID)
 }
 
+// Simple test to confirm that an existing genesis file will be deleted from the DB
+// TODO Confirm that the deletion of a very big file does not crash the machine
+func TestNodeNewNodeDeleteGenesisFileFromDB(t *testing.T) {
+	config := test.ResetTestRoot("node_new_node_delete_genesis_from_db")
+	defer os.RemoveAll(config.RootDir)
+	fmt.Println(config.RootDir)
+	// Use goleveldb so we can reuse the same db for the second NewNode()
+	config.DBBackend = string(dbm.GoLevelDBBackend)
+	// Ensure the genesis doc hash is saved to db
+	stateDB, err := cfg.DefaultDBProvider(&cfg.DBContext{ID: "state", Config: config})
+	require.NoError(t, err)
+
+	err = stateDB.SetSync(genesisDocKey, []byte("genFile"))
+	require.NoError(t, err)
+
+	genDocFromDB, err := stateDB.Get(genesisDocKey)
+	require.NoError(t, err)
+	require.Equal(t, genDocFromDB, []byte("genFile"))
+
+	stateDB.Close()
+	nodeKey, err := p2p.LoadOrGenNodeKey(config.NodeKeyFile())
+	require.NoError(t, err)
+
+	n, err := NewNode(
+		context.Background(),
+		config,
+		privval.LoadOrGenFilePV(config.PrivValidatorKeyFile(), config.PrivValidatorStateFile()),
+		nodeKey,
+		proxy.DefaultClientCreator(config.ProxyApp, config.ABCI, config.DBDir()),
+		DefaultGenesisDocProviderFunc(config),
+		cfg.DefaultDBProvider,
+		DefaultMetricsProvider(config.Instrumentation),
+		log.TestingLogger(),
+	)
+	require.NoError(t, err)
+
+	_, err = stateDB.Get(genesisDocKey)
+	require.Error(t, err)
+
+	// Start and stop to close the db for later reading
+	err = n.Start()
+	require.NoError(t, err)
+
+	err = n.Stop()
+	require.NoError(t, err)
+
+	stateDB, err = cfg.DefaultDBProvider(&cfg.DBContext{ID: "state", Config: config})
+	require.NoError(t, err)
+	genDocHash, err := stateDB.Get(genesisDocHashKey)
```

Comment:
Not sure if it's useful but a test like `ok, err := stateDB.Has(genesisDocHashKey)` could be useful?

Reply:
This might not make sense because of the issue with RocksDB